src/shared/install.c

```diff
@@ -3559,6 +3559,66 @@ int unit_file_get_list(
         return 0;
 }
 
+int unit_file_label_path(
+        UnitFileScope scope,
+        const char *root_dir,
+        const char *name,
+        char **ret_path) {
+
+        _cleanup_(lookup_paths_free) LookupPaths paths = {};
+        char **dirname;
+        int r;
+
+        assert(scope >= 0);
+        assert(scope < _UNIT_FILE_SCOPE_MAX);
+        assert(name);
+        assert_return(unit_name_is_valid(name, UNIT_NAME_ANY), -EINVAL);
+        assert(ret_path);
+
+        r = lookup_paths_init(&paths, scope, 0, root_dir);
```

Comment:
I can't say that I like this at all. `lookup_paths_init()` is already a fairly heavy operation. And filesystem lookup with directory scanning even more so. IIUC, we would get this overhead e.g. for all masked units, without any form of caching. Also, this duplicates the search logic, which is very finnicky, and it's likely that it does things differently than the existing search logic. So I don't think we can/should do this.

If it's necessary to look up paths for masked units, then instead please change the code that does the creation of the unit paths hashmaps to also return the paths for masked units somehow.

Reply:
So you prefer the previous implementation by keeping track of *real* unit files behind masked ones at load time, like https://github.com/systemd/systemd/pull/10023/commits/7672981d6e32d70672aa9da8e1762375a6987ee1
(more recent rebased version: https://github.com/systemd/systemd/pull/15245/commits/d1e8da4581f8038e3bf86e5c3c13fac5374d3b13)?

Reply:
I do not know what you mean but in case you're talking about SELinux labels here then please be aware that there is no such thing as a "fixed label". One should not hard code identifiers as they are supposed to be configurable by security policy.

Reply:
well, if there's no object to read a label off then we can't provide a label. so either it's a fixed label or we just don't do selinux here. you choose.

but the idea of going behind the mask to find a label just defeats the point of masking, that's a no go.

Reply:
Ive been thinking about these checks, and then it occurred to me that it is probably very easy to circumvent these checks anyway, as these operations essentially amount to just managing symlinks in /etc/systemd/system. Maybe I am overlooking something.